src/Utilities/VariableTemplates/VariableTemplates.jl

```diff
@@ -143,6 +144,7 @@ Grad{S}(array) where {S} = Grad{S,typeof(array),0}(array)
 
 Base.eltype(g::Grad) = eltype(getfield(g,:array))
 Base.propertynames(::Grad{S}) where {S} = fieldnames(S)
+Base.similar(g::Grad{S,A,offset}) where {S,A,offset} = Grad{S,A,offset}(similar(getfield(g,:array)))
```

Comment:
Is this tested in the tests? Maybe good to add.

Reply:
Good point.